pkg/operator/ceph/object/user/controller.go

```diff
@@ -310,14 +310,20 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
-	if store.Status != nil && store.Status.Info["secureEndpoint"] != "" {
-		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
+	if store.Status == nil {
+		return errors.Errorf("unknown store status while waiting for store to initialize")
 	}
-	if r.objContext.Endpoint == "" {
+
+	if store.Status.Info["endpoint"] != "" {
+		r.objContext.Endpoint = store.Status.Info["endpoint"]
+	} else {
```

Comment:
user can define `secure endpoint  without having `endpoint`. So this will error out in that case

Reply:
From what I can tell that is true in the spec, but in the status, `endpoint` is always defined. https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/status.go#L88.

Reply:
@alimaredia : this makes sense I missed if `secureport` only defined then `Status.Info[endpoint]` was filled instead of `Status.Info[secureendpoint]`. In the existing Store.Status nil check is missing I misread the situation